sourcegraph/sourcegraph
# Code Insights: Add backend insight widget

Closes https://github.com/sourcegraph/sourcegraph/issues/43192

## Problem 

It's described in the issue [here](https://github.com/sourcegraph/sourcegraph/issues/43192) but in a nutshell code insights is getting complex because of global shared FE types. By FE types I mean that these FE typescript types don't exist in API schema, we introduced them because our Code Insight API didn't have (and still doesn't have) a good types model over all types of insights). 

These FE types are bad because it shared across different Code Insights views and there is no easy way to update one view and don't touch all others. 

## Background

This is a part of big refactoring work about decoupling different parts of Code Insights web app, in this PR we decouple BackendInsightView (the dashboard card UI)

So the high-level picture about improvements for gql types is ([original comment about high level plan](https://github.com/sourcegraph/sourcegraph/issues/43192#issuecomment-1284521384))

1. Decouple global client based types (that every feature/page has its own client based types) [**This PR is doing this step for the backend insight card**]
2. Improve GQL API for a specific page/feature
3. Migrate this feature API from client based types to GQL types (feature by feature)
4. Go to step 2, repeat until we have no client types

- 8d3ecf9bfea07e5f1aac351e7fb4b224a51c29d4 Restructures the BackendView component and removes the polling feature flag. 
- aba1205ebee6fb72cc46e038bda91a12a41a4a59 Prior to this commit `BackendView` UI relied on shared (global) models, query, parsers and types, In this commit we pull everything that BackendView uses and put it closely to UI where it's used - `BackendView`
- 34704e28f07a0b88ab49418109c3e8f5ef6595bf In this Commit we lift the context menu UI because having it in the backend view UI forces us to have dashboard specific logic inside the `BackendView` (which is bad because it makes it hard to reuse)
- 2cd0423e637ab3454c69265628b55d41805db48a Same problem as with InsightContext UI but about dashboard specific API, we lift all dashboard API handlers on dashboard levels from `BackendView` UI

## Test plan
- Check that all unit and integration tests are green 
- Check that all drill down filters work as expected
- Check that filter creation flow (save as view) works correctly 
- Check that filters persist between page sessions 
- Check that context menu and all its options work as expected



## App preview:

- [Web](https://sg-web-vk-move-backend-insight-widget.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-rmqfktthkq.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
